src/inmanta/util.py

```diff
@@ -280,7 +280,7 @@ def remove_action_from_task_dict(task_dict: Dict[TaskMethod, List[asyncio.Task[o
     def add_action(
         self,
         action: TaskMethod,
-        schedule: TaskSchedule,
+        schedule: Union[TaskSchedule, int],  # int for backward compatibility,
```

Comment:
We could optionally make the two signatures a bit more clear by using `typing.overload`.

Reply:
never done that before, let me look into it.

Reply:
Example: https://github.com/inmanta/lsm/blob/26b2a4e33d1da5923bd1fc0adf30053be66e76e3/plugins/allocation_v2/dict_path.py#L384